pulsar-client-cpp/lib/ClientConnection.cc

```diff
@@ -941,13 +941,15 @@ ClientConnection::newConsumerStats(uint64_t consumerId, uint64_t requestId) {
 void ClientConnection::newTopicLookup(const std::string& destinationName, bool authoritative,
                                       const uint64_t requestId,
                                       LookupDataResultPromisePtr promise) {
+    Lock lock(mutex_);
```

Comment:
`outgoingCmd_` is being accessed outside of the lock in few places. eg: line 937